run_all_tests.py

```diff
@@ -4,19 +4,40 @@
 from twisted.python.runtime import platformType
 from twisted.scripts.trial import _makeRunner, Options, _initialDebugSetup
 
-from testlist import ipv8_testsuite
+from unittest import TestLoader
+import unittest
+
+def cleanup_testsuite(arg):
```

Comment:
Usually test suites only include methods which start with `test_`.

Reply:
disable the cleaner and witness the effect :wink:

Reply:
Why are you not doing the same?

Reply:
```python
Running 367 tests.
ipv8.test.base
  TestBase
    runTest ...                                                            [OK]
ipv8.test.dht.test_routing
  TestBucket
    test_add_update ...                                                    [OK]
    test_get ...                                                           [OK]
Unhandled Error
Traceback (most recent call last):
  File "/home/vader/.local/lib/python2.7/site-packages/twisted/protocols/amp.py", line 1043, in _commandReceived
    deferred = self.dispatchCommand(box)
  File "/home/vader/.local/lib/python2.7/site-packages/twisted/protocols/amp.py", line 1101, in dispatchCommand
    return maybeDeferred(responder, box)
  File "/home/vader/.local/lib/python2.7/site-packages/twisted/internet/defer.py", line 150, in maybeDeferred
    result = f(*args, **kw)
  File "/home/vader/.local/lib/python2.7/site-packages/twisted/protocols/amp.py", line 1188, in doit
    return maybeDeferred(aCallable, **kw).addCallback(
--- <exception caught here> ---
  File "/home/vader/.local/lib/python2.7/site-packages/twisted/internet/defer.py", line 150, in maybeDeferred
    result = f(*args, **kw)
  File "/home/vader/.local/lib/python2.7/site-packages/twisted/trial/_dist/worker.py", line 77, in addSuccess
    self._result.addSuccess(self._testCase)
  File "/home/vader/.local/lib/python2.7/site-packages/twisted/trial/_dist/distreporter.py", line 83, in addSuccess
    self.running[test.id()].append((self.original.addSuccess, test))
exceptions.KeyError: 'ipv8.test.base.TestBase.runTest'

                                                        [FAIL]
```

That's why.

Reply:
Or what was your question? Why don't I check for "test_" there?

Reply:
Yes

Reply:
Because detecting something based on names is not reliable. It is more reliable to mark the class we want to exclude by adding some property/method.

Reply:
Name-based detection is the reason why we even got this problem in the first place.

Reply:
I'm not sure if we're talking about the same thing. What I'm trying to say is that the default behavior of all available test suites is to only `addTest()` methods from test classes which start with `test_`. This is what programmers expect. What is unexpected is methods like `runTest()` are being executed.

Reply:
Yes, it is unexpected. It is a `unittests` bug. However, I don't like the name-based detection method in the first place. I think it is bad design. I would expect this method to be used in `bash`-style scripting languages, and I see no reasons why any language with proper module import system, like Python, would use this.
That's why I didn't use the text-based detection in my fix. Because I think this kind of design is wrong.
However, it's your code. If you insist that the cleaner's detection method should be implemented with text-scanning, I can change it.